scan/tls_handshake.go

```diff
@@ -30,8 +54,9 @@ func sayHello(host string, ciphers []uint16, vers uint16) (cipherIndex int, err
 	config.MinVersion = vers
 	config.MaxVersion = vers
 	config.CipherSuites = ciphers
+	config.CurvePreferences = curves
 	conn := tls.Client(tcpConn, config)
-	serverCipher, serverVersion, err := conn.SayHello()
+	serverCipher, serverCurveType, serverCurve, serverVersion, err := conn.SayHello()
 	conn.Close()
 	if err != nil {
 		return
```

Comment:
So you don't think there's value in logging any of the other errors that can happen during readHandshake? There are, for example, interesting errors that can be returned from  `conn.readRecord` and it might be worth getting them out to the user - shadowing them will make the tool less useful for debugging odd behavior. Or am I overthinking this?